binding/vue/README.md

```diff
@@ -110,89 +110,195 @@ const porcupineModel = {
-    console.log(isLoaded);
-  },
-  isListeningCallback: function(isListening) {
-    console.log(isListening);
-  },
-  errorCallback: function(error) {
-    console.error(error);
-  }
-};
-...
-```
-
-Import `Porcupine` mixin, add it to your component and initialize Porcupine:
-
-```html
-<script lang="ts">
-  import {BuiltInKeyword} from "@picovoice/porcupine-web";
-  import porcupineMixin from "@picovoice/porcupine-vue";
-  
-  import porcupineParams from "${PATH_TO_PORCUPINE_PARAMS_BASE64}"
-
-  export default {
-    mixins: [porcupineMixin],
-    mounted() {
-      this.$porcupine.init(
-              ${ACCESS_KEY},
-              [BuiltInKeyword.Porcupine],
-              this.keywordDetectionCallback,
-              { base64: porcupineParams }, // porcupine model
-              this.isLoadedCallback,
-              this.isListeningCallback,
-              this.errorCallback,
-      );
+await this.start();
+```
+
+If `WebVoiceProcessor` has started correctly, `state.isListening` will be set to true. Watch `state.keywordDetection` to get keyword detection results.
+
+#### Stop
+
+Run `stop` to stop keyword detection:
+
+```typescript
+await this.stop();
+```
+
+If `WebVoiceProcessor` has stopped correctly, `state.isListening` will be set to false.
+
+#### Release
+
+Run release explicitly to clean up all resources used by `Porcupine` and `WebVoiceProcessor`:
+
+```typescript
+this.release();
+```
+
+This will set `state.isLoaded` and `state.isListening` to false.
+
+### Porcupine in Vue 3
+
+For Vue 3, we take advantage of the [Composition API](https://vuejs.org/api/composition-api-setup.html), specially use of `reactive`.
+
+Use `usePorcupine` and `init` to initialize `Porcupine`:
+
+```vue
+
+<script lang='ts'>
+import { defineComponent, onBeforeUnmount, watch } from 'vue';
+import { BuiltInKeyword } from '@picovoice/porcupine-web';
+import { usePorcupine } from '@picovoice/porcupine-vue';
+
+// Use Vue.extend for JavaScript
+export default defineComponent({
+  setup() {
+    const { 
+      state,
+      init,
+      start,
+      stop,
+      release
+    } = usePorcupine();
+    
+    watch(() => state.isLoaded, (newVal) => {
+      console.log(newVal);
+    });
+
+    watch(() => state.isListening, (newVal) => {
+      console.log(newVal);
+    });
+
+    watch(() => state.keywordDetection, (keyword) => {
+      if (keyword !== null) {
+        console.log(keyword.label);
+      }
+    });
+    
+    watch(() => state.error, (err) => {
+      if (err) {
+        console.error(err);
+      }
+    });
+
+    onBeforeUnmount(() => {
+      release();
+    });
+    
+    init(
+      ${ACCESS_KEY},
+      [BuiltInKeyword.Porcupine],
+      porcupineModel
+    )
+    
+    return {
+      start,
+      stop,
+      release
     }
   }
+});
 </script>
 ```
 
-### Process Audio Frames
+In case of any errors, watch for `state.error` to check the error message, else watch `state.isLoaded` to check if `Porcupine` has loaded.
```

Comment:
```suggestion
In case of any errors, watch for `state.error` to check the error message, otherwise watch `state.isLoaded` to check if `Porcupine` has loaded.
```